sorting/odd_even_sort.c

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file
+ * @author [Edwin Ajong](https://github.com/eddybruv)
+ * @brief [Odd Even Sort](https://en.wikipedia.org/wiki/Odd%E2%80%93even_sort) implementation
+ * @details
+ * This algorithm is divided into two phases- Odd and Even Phase.
+ * The algorithm runs until the array elements are sorted and in each iteration two phases occurs- Odd and Even Phases.
+ * In the odd phase, we perform a bubble sort on odd indexed elements and in the even phase,
+ * we perform a bubble sort on even indexed elements.
+ * Time Complexity: O(N ^ 2)
+ */
+
+#include <assert.h>     /// for assert
+#include <stdbool.h>    /// for bool
+#include <stdio.h>      /// for IO operations
+#include <stdlib.h>     /// for dynammic memory allocation
+#include <time.h>       /// for random number generation
+
+/**
+ * @brief Swap numbers by reference(using pointers)
+ * @param first pointer to first number
+ * @param second pointer to second number
+ * @returns void
+ */
+
+void swap(int *first, int *second)
+{
+    int temp = *first;
```

Comment:
Consider using `uint64_t` for non-negative values (or their appropriate size: `uint32_t`, `uint16_t`, `uint8_t`) or `int64_t` for negative values. **Requires adding the `inttypes.h` header.** Check other parts of the code ([reference](https://en.cppreference.com/w/c/types/integer)). 🙂